R/environment_shadow.r

```diff
@@ -2,8 +2,15 @@
 # This is needed to build in our own needs, like properly shutting down the kernel
 # when `quit()` is called.
 
-add_to_user_searchpath <- function(name, FN) {
-    assign(name, FN, 'jupyter:irkernel')
+add_to_user_searchpath <- function(name, FN, pkg = NULL) {
+    if (is.null(pkg)) {
+        assign(name, FN, 'jupyter:irkernel')
+    } else {
+        env <- getNamespace(pkg)
```

Comment:
what happens if the package isn’t installed?

please make sure this case is handled by falling back to the `'jupyter:irkernel'` environment.

It would also be great if you could add tests for `add_to_user_searchpath` that cover all three usages (without `pkg`, with existing `pkg`, and with non-existing `pkg`